src/binder/clrprivbinderassemblyloadcontext.cpp

```diff
@@ -251,9 +271,51 @@ HRESULT CLRPrivBinderAssemblyLoadContext::SetupContext(DWORD      dwAppDomainId,
     return hr;
 }
 
+/* static */
+void CLRPrivBinderAssemblyLoadContext::DestroyContext(CLRPrivBinderAssemblyLoadContext *pBindContext, INT_PTR ptrManagedStrongAssemblyLoadContext)
+{
+    CONTRACTL
+    {
+        GC_NOTRIGGER;
+        THROWS;
+        MODE_PREEMPTIVE;
+        SO_TOLERANT;
+    }
+    CONTRACTL_END;
+
+    GCX_COOP();
```

Comment:
If we are going to switch to COOP mode immediately, then please update the contract to MODE_COOPERATIVE and make the callsites perform the mode transition instead of it happening here.